presto-main/src/main/java/io/prestosql/sql/planner/plan/GroupIdNode.java

```diff
@@ -67,6 +67,9 @@ public GroupIdNode(
         this.aggregationArguments = ImmutableList.copyOf(aggregationArguments);
         this.groupIdSymbol = requireNonNull(groupIdSymbol);
 
+        checkArgument(source.getOutputSymbols().containsAll(groupingColumns.values()));
+        checkArgument(source.getOutputSymbols().containsAll(aggregationArguments));
```

Comment:
Can this be a perf problem for wide tables?
`ImmutableSet.copyOf(source.getOutputSymbols())` would be safer.